src/Backups/BackupSettings.h

```diff
@@ -90,6 +90,13 @@ struct BackupSettings
     /// Is it allowed to use blob paths to calculate checksums of backup entries?
     bool allow_checksums_from_remote_paths = true;
 
+    /// Generator for a unique and deterministic backup file name.
+    /// If specified, should be one of ["first_file_name", "checksum"]. Defaults to "first_file_name".
+    String data_file_name_generator;
+
+    /// A length of backup file name prefix. Only applicable to "checksum" generator.
+    std::optional<size_t> data_file_name_prefix_length;
```

Comment:
`std::nullopt` seems meaningless here so I suggest using here just
```
size_t data_file_name_prefix_length = 3;
```

Reply:
Non-optional has meaning only for checkum generator.
Besides, it will become impossible to override through `backups. data_file_name_prefix_length` config setting, so I will keep it as `std::optional`.